apps/react-playground/src/app/Widgets/MessagingPrototype/Messaging.tsx

```diff
@@ -0,0 +1,178 @@
+          }}
+        >
+          step 3:Now that we have our authorize derive key transaction
+          constructed and enough DESO lets submit it
+        </button>
+      </div>
+      <div>
+        <button
+          className={buttonClass}
+          onClick={async () => {
+            console.log('generate the default key');
+            console.log(deso.identity.getUserKey());
+            messagingGroupPayload = await deso.identity.messagingGroups(
+              deso.identity.getUserKey() as string
+            );
+          }}
+        >
+          step 4: generate the default key
+        </button>
+      </div>
+
+      <div className={containerClass}>
+        <button
+          className={buttonClass}
+          onClick={() => {
+            console.log('encrypt');
+
+            // encryptedApplicationGroupMessagingKey: string,
+            // applicationSeedHex: string,
+            // recipientPublicKey: string,
+            // message: string
+            if (messagingGroupPayload) {
+              const encryptedMessage =
+                encryptMessageFromEncryptedToApplicationGroupMessagingKey(
+                  messagingGroupPayload.encryptedToApplicationGroupMessagingPrivateKey,
+                  deso.utils.privateKeyToSeedHex(
+                    messagingGroupPayload.encryptedToApplicationGroupMessagingPrivateKey
+                  ),
+                  USER_TO_SEND_MESSAGE_TO,
+                  'message to encrypt'
+                );
+              console.log(encryptedMessage);
+            }
+          }}
+        >
+          step 5: encrypt
+        </button>
+      </div>
+
+      <div className={containerClass}>
+        <button
+          className={buttonClass}
+          onClick={() => {
+            if (messagingGroupPayload) {
+              decryptMessageFromEncryptedToApplicationGroupMessagingKey(
+                messagingGroupPayload.encryptedMessagingKeyRandomness ?? '',
+                deso.utils.privateKeyToSeedHex(
+                  messagingGroupPayload.encryptedToApplicationGroupMessagingPrivateKey
+                ),
+                {
+                  EncryptedHex:
+                    '04dc277d8ba1ad4116ea9143d27a92be5d520de5f5c9d4645f85fe5f6b5791e5be8c8a06bd563c86e557e870ef8c94124102df17ab16548476336c6769cde095cf2758eefc56f9d1cdccf8a6d1b0a5dc6d09054067306a99666783ea36df93d03e81cd2d2ab3a32112c3a5db0630d3bed806a647',
+                  IsSender: false,
+                  Version: 2,
+                  SenderMessagingPublicKey:
+                    'BC1YLjMYu2ahUtWgSX34cNLeM9BM9y37cqXzxAjbvPfbxppDh16Jwog',
+                  SenderMessagingGroupKeyName: '',
+                  RecipientMessagingPublicKey:
+                    'BC1YLheA3NepQ8Zohcf5ApY6sYQee9aPJCPY6m3u6XxCL57Asix5peY',
+                  RecipientMessagingGroupKeyName: '',
+                  PublicKey: deso.identity.getUserKey() as string,
+                  Legacy: false,
+                }
+              );
```

Comment:
I'm assuming this is just for testing purposes, correct? We'll want to actually hook it up to `get-messages-stateless`. Perhaps we update `get-messages-stateless` to accept some sort of function to decrypt instead of using identity?

Reply:
and we'll probably want to display the decrypt message threads somewhere here. like something similar to the messaging inbox component we have in the reference frontend implementation

Reply:
Yeah just throwing in some dummy data until I hook up real messages